### PR TITLE
implement saving and sending TranscriptResult information in JSON

### PR DESCRIPTION
**Pull request by nikvaessen**
Relies on #70 

Example chat message:
```
{
  "transcript":[
    {
      "confidence":0.9876291155815125,
      "text":"how are you"
    }
  ],
  "name":"Nik",
  "is_interim":false,
  "language":"en-US",
  "message_id":"ab93f8d1-4120-4296-8120-7b4f80db0e22",
  "id":"c02d8481",
  "event":"SPEECH",
  "timestamp":"2017-08-07T14:26:23.637Z"
}
```

Example complete transcript:

```
{
  "room_name": "test123@conference.nik.jitsi.net",
  "start_time": "2017-08-07T14:31:55.110Z",
  "initial_participants": [
    "Nik"
  ],
  "end_time": "2017-08-07T14:32:21.333Z",
  "events": [
    {
      "transcript": [
        {
          "confidence": 0.932471752166748,
          "text": "hey how are you"
        }
      ],
      "name": "Nik",
      "is_interim": false,
      "language": "en-US",
      "message_id": "a235ba29-a966-4e8b-a9fa-95b034fbffac",
      "id": "c02d8481",
      "event": "SPEECH",
      "timestamp": "2017-08-07T14:32:05.848Z"
    },
    {
      "transcript": [
        {
          "confidence": 0.9122274518013,
          "text": "I love summer of goats"
        }
      ],
      "name": "Nik",
      "is_interim": false,
      "language": "en-US",
      "message_id": "669b11ae-e0a3-4887-acc8-c391bdfb3468",
      "id": "c02d8481",
      "event": "SPEECH",
      "timestamp": "2017-08-07T14:32:10.382Z"
    },
    {
      "transcript": [
        {
          "confidence": 0.7015020847320557,
          "text": "my"
        }
      ],
      "name": "Nik",
      "is_interim": false,
      "language": "en-US",
      "message_id": "89031986-3eb5-4a6e-8a09-368cd1657918",
      "id": "c02d8481",
      "event": "SPEECH",
      "timestamp": "2017-08-07T14:32:18.309Z"
    }
  ]
}
```

### CODE REVIEW

File: src/main/java/org/jitsi/jigasi/TranscriptionGateway.java [hunk 1]
```diff
@@ -50,9 +50,11 @@ public void stop()
...
     {
         TranscriptionGatewaySession outgoingSession =
-                new TranscriptionGatewaySession(this, ctx,
-                        new GoogleCloudTranscriptionService(),
-                        new LocalTxtTranscriptHandler());
+                new TranscriptionGatewaySession(
+                    this,
+                    ctx,
+                    new GoogleCloudTranscriptionService(),
+                    new LocalJsonTranscriptHandler());
```
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/TranscriptionGateway.java]: How to make this configurable?

**Comment by bgrozev** [comment]
What do you think about moving an event's "name" and "id" fields to a "participant" object?
```json
participant: {"name": "Nik", "id": "0abcd123"}
```
We can also add this to the initial participants' array (since names can be spoofed).


> I love summer of goats

LOL. GSoG?

**Comment by nikvaessen** [comment]
> What do you think about moving an event's "name" and "id" fields to a "participant" object?
added :)

> LOL. GSoG?

I think Google is secretly thinking about a name change!

File: src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jigasi.transcription;
+
+import com.google.gson.*;
```
Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: Which maven package does this come from?

Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: https://mvnrepository.com/artifact/com.google.code.gson/gson

But it's not imported in jigasi's pom.xml, so can it be a from another dependency?

Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: That's what I'm guessing. Is it a separate library from org.json.simple? Why do we use both of them?

Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: There's actually 3 json libraries imported: org.json.simple, [org.json.json](https://mvnrepository.com/artifact/org.json/json) and com.google.gson. 

I use google's to pretty-print the json to the transcript whicn can't be done with the json-simple as far as I know

File: src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    }
+
+    @Override
+    public void publish(JSONObject transcript)
+    {
+        String uglyJSON = transcript.toJSONString();
+
+        String prettyJSOn = new GsonBuilder().setPrettyPrinting().create().
+            toJson(new JsonParser().parse(uglyJSON));
+
```
Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: this is the only place where gson is used, the rest is json.simple

Comment (bgrozev) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: Oh, is the purpose just to add whitespace for prettier formatting? If so, can we remove it - -we'll do that handling in the client.

Comment (nikvaessen) [inline src/main/java/org/jitsi/jigasi/transcription/LocalJsonTranscriptHandler.java]: I will remove it - but it makes debugging harder as it makes it impossible to read the final json transcript

**Pull request merged**